examples/vision/xray_classification_with_tpus.py

```diff
@@ -0,0 +1,489 @@
+    # load the raw data from the file as a string
+    img = tf.io.read_file(file_path)
+    img = decode_img(img)
+    return img, label
+
+
+train_ds = train_list_ds.map(process_path, num_parallel_calls=AUTOTUNE)
+
+val_ds = val_list_ds.map(process_path, num_parallel_calls=AUTOTUNE)
+
+"""
+Let's visualize the shape of an (image, label) pair.
+"""
+
+for image, label in train_ds.take(1):
+    print("Image shape: ", image.numpy().shape)
+    print("Label: ", label.numpy())
+
+"""
+Load and format the test data as well.
+"""
+
+test_list_ds = tf.data.Dataset.list_files(str(GCS_PATH + "/chest_xray/test/*/*"))
+TEST_IMAGE_COUNT = tf.data.experimental.cardinality(test_list_ds).numpy()
+test_ds = test_list_ds.map(process_path, num_parallel_calls=AUTOTUNE)
+test_ds = test_ds.batch(BATCH_SIZE)
+
+TEST_IMAGE_COUNT
+
+"""
+# 3. Visualize the dataset
+
+First, let's use buffered prefetching so we can yield data from disk without having I/O
+become blocking.
+"""
+
+
+def prepare_for_training(ds, cache=True, shuffle_buffer_size=1000):
+    # This is a small dataset, only load it once, and keep it in memory.
+    # use `.cache(filename)` to cache preprocessing work for datasets that don't
+    # fit in memory.
+    if cache:
+        if isinstance(cache, str):
+            ds = ds.cache(cache)
+        else:
+            ds = ds.cache()
+
+    ds = ds.shuffle(buffer_size=shuffle_buffer_size)
+
+    # Repeat forever
+    ds = ds.repeat()
+
+    ds = ds.batch(BATCH_SIZE)
+
+    # `prefetch` lets the dataset fetch batches in the background while the model
+    # is training.
+    ds = ds.prefetch(buffer_size=AUTOTUNE)
+
+    return ds
+
+
+"""
+Call the next batch iteration of the training data.
+"""
+
+train_ds = prepare_for_training(train_ds)
+val_ds = prepare_for_training(val_ds)
+
+image_batch, label_batch = next(iter(train_ds))
+
+"""
+Define the method to show the images in the batch.
+"""
+
+
+def show_batch(image_batch, label_batch):
+    plt.figure(figsize=(10, 10))
+    for n in range(25):
+        ax = plt.subplot(5, 5, n + 1)
+        plt.imshow(image_batch[n])
+        if label_batch[n]:
+            plt.title("PNEUMONIA")
+        else:
+            plt.title("NORMAL")
+        plt.axis("off")
+
+
+"""
+As the method takes in numpy arrays as its parameters, call the numpy function on the
+batches to return the tensor in numpy array form.
+"""
+
+show_batch(image_batch.numpy(), label_batch.numpy())
+
+"""
+# 4. Build the CNN
+
+To make our model more modular and easier to understand, let's define some blocks. As
+we're building a convolution neural network, we'll create a convolution block and a dense
+layer block.
+
+The architecture for this CNN has been inspired by this
+[article](https://towardsdatascience.com/deep-learning-for-detecting-pneumonia-from-x-ray-images-fc9a3d9fdba8).
+[article](https://towardsdatascience.com/deep-learning-for-detecting-pneumonia-from-x-ray-images-fc9a3d9fdba8).
+"""
+
+
+def conv_block(filters):
```

Comment:
Prefer using the functional API, e.g.

```python
inputs = keras.Input(shape=(IMAGE_SIZE[0], IMAGE_SIZE[1], 3))
x = preprocessing.Rescaling(1./255)(inputs)
x = layers.Conv2D(16, 3, activation="relu", padding="same")(x)
x = layers.Conv2D(16, 3, activation="relu", padding="same")(x)
...
x = conv_block(x, 32)
etc
```

Reply:
I left the `conv_block` and `dense_block` as is and changed the `build_model` to the functional API. Please let me know if I should change the blocks as well